crypto/merklearray/worker.go

```diff
@@ -25,6 +25,11 @@ import (
 // workerState describes a group of goroutines processing a sequential list
 // of maxidx elements starting from 0.
 type workerState struct {
+	// maxidx is the total number of elements to process, and nextidx
```

Comment:
Do these other cases also need a similar treatment? 

```
./ledger/acctupdates.go 388:	return atomic.LoadInt32(&au.catchpointWriting) != 0
./ledger/bulletin.go 42:	if atomic.CompareAndSwapUint32(&notifier.notified, 0, 1) {
./util/metrics/segment.go 54:	concurrentCounter := atomic.AddUint32(&segment.concurrentCounter, uint32(1))
./util/metrics/segment.go 84:	concurrentCounter := atomic.AddUint32(&segInstance.segment.concurrentCounter, ^uint32(0))
./network/wsNetwork.go 1225:		if curPeersChangeCounter := atomic.LoadInt32(&wn.peersChangeCounter); curPeersChangeCounter != lastPeersChangeCounter {
./network/wsNetwork.go 2053:	if atomic.AddInt32(&wn.throttledOutgoingConnections, int32(-1)) >= 0 {
./network/wsNetwork.go 2228:		if atomic.CompareAndSwapInt32(&wn.ready, 0, 1) {
./network/wsNetwork_test.go 158:	if atomic.LoadInt32(&mch.shouldWait) > 0 {
./network/wsNetwork_test.go 752:	return atomic.LoadInt32(&peer.didInnerClose) != 0
./network/wsPeer.go 366:	if atomic.LoadInt32(&wp.didInnerClose) == 0 {
./network/wsPeer.go 610:		if atomic.LoadInt32(&wp.didInnerClose) == 0 {
./catchup/service.go 129:	if atomic.CompareAndSwapUint32(&s.initialSyncNotified, 0, 1) {
./logging/telemetryOperation.go 41:	if !atomic.CompareAndSwapInt32(&op.pending, 1, 0) {
```

Reply:
I'm pretty sure that the same was already applied to the rest of them. We generally make these change pretty quickly since the failure is very straight-forward.